tests/trainer/test_config_validator.py

```diff
@@ -147,3 +147,17 @@ def predict_dataloader(self):
 
     with pytest.raises(MisconfigurationException, match="Dataloader not found for `Trainer.predict`"):
         trainer.predict(model)
+
+
+def test_trainer_manual_optimization_config(tmpdir):
+    """ Test error message when requesting Trainer features unsupported with manual optimization """
+    model = BoringModel()
+    model.automatic_optimization = False
+
+    with pytest.raises(MisconfigurationException, match="Automatic gradient clipping is not supported"):
+        trainer = Trainer(gradient_clip_val=1.0)
```

Comment:
nit: The trainer creation should be out of the with block

Reply:
addressed the flaw